agents/rust/mediator/src/didcomm_handlers/mod.rs

```diff
@@ -47,13 +46,21 @@ pub async fn handle_aries<T: BaseWallet + 'static, P: MediatorPersistence>(
             handle_routing_forward(agent.clone(), forward).await?;
             return Ok(Json(json!({})));
         } else {
-            let (account_name, our_signing_key, their_diddoc) =
+            // Auth known VerKey then process account related messages
+            let (account_name, auth_pubkey, our_signing_key, their_diddoc) =
                 agent.auth_and_get_details(&unpacked.sender_verkey).await?;
-            let auth_pubkey = unpacked
-                .sender_verkey
-                .expect("Sender key authenticated above, so it must be present..");
             log::info!("Processing message for {:?}", account_name);
             match aries_message {
+                GeneralAriesMessage::AriesVCXSupported(AriesMessage::Pickup(pickup_message)) => {
+                    let pickup_response =
+                        handle_pickup_protocol(&agent, pickup_message, &auth_pubkey).await?;
+                    let aries_response = AriesMessage::Pickup(pickup_response);
+                    let aries_response_bytes =
+                        serde_json::to_vec(&aries_response).map_err(string_from_std_error)?;
+                    agent
+                        .pack_didcomm(&aries_response_bytes, &our_signing_key, &their_diddoc)
+                        .await?
+                }
```

Comment:
bit of a duplication between the `GeneralAriesMessage::AriesVCXSupported(AriesMessage::Pickup(pickup_message))` and `GeneralAriesMessage::XumCoord(coord_message)` branches, maybe could be refactored?

Reply:
I looked into this. Currently, I don't see a way to. 

Since both the structs are of different types from different sources, there's subtle difference in how they need to be serialised, which is mostly what is going on there. 

I could join the flow immediately after serialising, and have common packing, but that would offer very little advantage..

Once MediatorCoord protocol messages are part of `aries-vcx/messages` this should be easier to do.

Reply:
ah yeah, i see what you mean. Yeah this should be possible to simplify later